src/ws/service/friendships_service.rs

```diff
@@ -100,18 +98,53 @@ impl FriendshipsServiceServer<SocialContext> for MyFriendshipsService {
             Err(err) => {
                 // TODO: Handle error when trying to get User Id.
                 log::debug!("Get Friends > Get User ID from Token > Error: {err}.");
-                // Err(FriendshipsError::CommonError(CommonError::Unknown)),
-                let (g, _) = Generator::create();
-                g
+                // Err(FriendshipsError::CommonError(CommonError::err)),
+                todo!()
             }
         }
     }
     async fn get_request_events(
         &self,
-        _request: Payload,
-        _context: Arc<SocialContext>,
+        request: Payload,
+        context: Arc<SocialContext>,
     ) -> RequestEvents {
-        todo!()
+        // Get user id with the given Authentication Token.
+        let user_id = get_user_id_from_request(&request, &context).await;
+
+        match user_id {
+            Ok(user_id) => {
+                // Look for users requests
+                match context.app_components.db.db_repos.clone() {
+                    Some(repos) => {
+                        let requests = repos
+                            .friendship_history
+                            .get_user_request_events(&user_id.social_id)
+                            .await;
+                        match requests {
+                            // TODO: Handle get user requests query response error.
+                            Err(err) => {
+                                log::debug!("Get Friends > Get User Requests > Error: {err}.");
+                                // Err(FriendshipsError::CommonError(CommonError::Unknown)),
+                                todo!()
+                            }
+                            Ok(it) => map_request_events(it, user_id.social_id),
```

Comment:
Can we rename the var to something more declarative? As it's not an iterator, it's a vector of requests
```suggestion
                            Ok(requests) => map_request_events(requests, user_id.social_id),
```